frontend/src/app/matching/matching.component.html

```diff
@@ -0,0 +1,63 @@
+<div class="container">
+    <div class="form-wrapper w-full sm:w-30rem">
+        <h2 class="mt-0 mb-1">Matching Criteria</h2>
+        <p class="mt-0">Select any one criteria to start matching!</p>
+        <form #matchForm="ngForm" class="flex flex-column gap-3">
+            <div class="form-field">
+                <label for="topic">Topic</label>
+                <p-multiSelect
+                    name="topic"
+                    [options]="topics"
+                    [(ngModel)]="userCriteria.topics"
+                    placeholder="No Topic Selected"
+                    [loading]="isLoadingTopics"
+                    [showClear]="true"
+                    [maxSelectedLabels]="0"
+                    selectedItemsLabel="{{ userCriteria.topics?.length }} topic{{
+                        userCriteria.topics !== null && userCriteria.topics.length > 1 ? 's' : ''
+                    }} selected"
+                    class="p-fluid" />
+
+                @if (userCriteria.topics !== null && userCriteria.topics.length > 0) {
+                    <div class="flex flex-wrap gap-2">
+                        <p-chip
+                            *ngFor="let topic of userCriteria.topics; let i = index"
+                            [label]="topic"
+                            removable="true"
+                            (onRemove)="removeTopic(i)"
+                            styleClass="text-sm" />
+                    </div>
+                }
+            </div>
+
+            <div class="form-field">
+                <label for="difficulty">Difficulty</label>
+                <p-dropdown
+                    name="difficulty"
+                    [options]="difficulties"
+                    [(ngModel)]="userCriteria.difficulty"
+                    placeholder="No Difficulty Selected"
+                    [showClear]="true"
+                    class="p-fluid" />
+            </div>
+
+            <p-button
+                type="submit"
+                [label]="isProcessingMatch ? '' : 'Start Matching'"
+                [loading]="isProcessingMatch"
+                [disabled]="userCriteria.topics === null && userCriteria.difficulty === null"
```

Comment:
Perhaps you could initialize `userCriteria.topics` as `[]` and check if `userCriteria.topics.length === 0`?

Reply:
Weird thing is that when I clear the input or deselect all options via the PrimeNG component, the value of topics gets set to null. So I just opted to set null as the default state

Reply:
I think I'll just use the `onClear` event to set topics to `[]` instead.